proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+- Phab reviewers list
+
+  Reviewers are currently assigned based on a set of Phabricator rules. As a maintainer on a less popular platform I know the pain of getting people to review your code.
+  
+  So my suggestion is to also assign reviewers based on the git history of the files you're changing. 
+
+  The reason for this is that currently it's always the same people reviewing patches.
+  Their time is spread thin. Particularly on less popular platforms it basically comes down to 4 people.
+  
+- Update trac linters
+
+  The git linters and Arc linters aren't currently checking the same things.
+
+  Particularly is the fact that the pre-commit hooks check
+  the summary text but arc doesn't. So commits can get rejected at push time requiring more fork for maintainers.
+
+  Also I want to say I love the summary document you have to fill in.
+  It ensures useful information is there later when I have to find out why a change was made. So whatever we do, don't remove this.
+
+- Phab signup screen
+
+  It's recently been made a requirement to require a public key to push to phab. The error you get when you don't do this and try to push a patch is very very cryptic and unintuitive. Could we make a plugin that asks the user to upload a public key on trac if they haven't done so? Like a banner at the top?
+
+- Add some automation to Trac (Or whatever bug tracker we end up using)
+
+  - Particular on new tickets post a friendly reminder that if they want they can give it a hand in fixing it themselves.
+  - Parse information added, in particular check if reproduction steps are there etc.
+  - If stack is used, kindly ask if a repo without can be used. The amount of bug reports with stack is increasing and regardless of my own opinion on the tool, these reports are not very useful as is.
+  - Maybe automatically CC people from a pool based on the information in the ticket? I tend to miss tickets because my filters are quite strict. Generally if the ticket doesn't mention my name, is directed at me or has "Windows" in the body somewhere it will skip my inbox. I review filtered tickets only once a week.
+  - If a newcomer assigns a ticket to themselves, have trac automatically post links to useful pages:
+  
+      - how to setup build environment.
+      - how to get help.
+      - assign a mentor?
+      - after x amount of time with no progress, remind them again that help is available
+
+- Unify the issue tracker and code review into phabricator
+
+- Move the Wiki someplace else. Is the wiki in phabricator any good, maybe we should use it.
+  However, finding something in the ghc wiki has been more hit and miss for me than anything else. The
+  built in search almost never reveals what I’m looking for, and resorting to a search engine only sometimes
+  helps me find what I’m looking for. (This might totally be my fault).
+
+- Reduce the number of Mailing lists. Some people are supposedly not on the haskell ml; so they missed emails such as
```

Comment:
Really GHC only has two mailing lists: `glasgow-haskell-users` and `ghc-devs`. I think this is a pretty sensible split. I think any coarsening would serve only to decrease signal to noise (especially for users, who have little interest in GHC internals).